src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,169 @@
+package util;
+
+import backend.Logic;
+import backend.github.ApiQuotaInfo;
+import org.apache.logging.log4j.Logger;
+import ui.UI;
+import util.events.RefreshTimerTriggeredEvent;
+import util.events.RefreshTimerTriggeredEventHandler;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages the API quota usage and periodic refreshes of the data store.
+ */
+public class ApiQuotaManager {
+
+    private static final Logger logger = HTLog.get(ApiQuotaManager.class);
+
+    /**
+     * The amount that is set aside for manual operations by the user.
+     */
+    private static final int API_QUOTA_BUFFER = 200;
+
+    /**
+     * The default refresh period.
+     */
+    private static final int DEFAULT_REFRESH_PERIOD_IN_MINS = 1;
+
+    /**
+     * The previous amount of the available remaining API requests.
+     */
+    private int previousRemainingApiRequests = 0;
```

Comment:
I see what you mean. To me, it's less confusing if you prefix the other one (e.g. `newRemainingApiRequests`), which has a smaller scope. 
I think `difference` is not very meaningful. How about,
`apiUsageInRecentRefresh = remainingApiRequests - newRemainingApiRequests`
It is also a bit odd that this variable is set to 0 at the beginning. That doesn't match the reality, right?